app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -227,50 +294,51 @@ def _validate_single_branch_for_rollout(self, data):
         return data
 
     def update(self, experiment, data):
-        with transaction.atomic():
-            reference_branch_data = data.pop("reference_branch", None)
-            treatment_branches_data = data.pop("treatment_branches", None)
+        data.pop("reference_branch", None)
```

Comment:
Good clean refactor here. :+1: Scratching my head at why this was complicated originally. IIRC, I think [this bugfix for on_delete=SET_NULL](https://github.com/mozilla/experimenter/commit/7385e50dbc0a2651b8ffb349da37c4d41b6c5b8f#diff-36bd393fce70ef5fc7ad821bc3c232a4b00882f000583f222fc3041f735d3bdbR156) needed to land first. The experiment itself would get deleted without explicitly setting `experiment.reference_branch = None` first. But, this works now so yay!